zkvm/host/src/backend/zisk.rs

```diff
@@ -0,0 +1,311 @@
+            ));
+        }
+
+        // Extract the execution cycle from the output.
+        let cycles: u64 = Self::extract_cycles(&output)?;
+
+        // Read the public output from VM guest.
+        let state_root = fs::read(&output_path).map_err(|e| {
+            anyhow!(
+                "Couldn't read from execution output. Error:\n{}",
+                e.to_string()
+            )
+        })?;
+
+        Ok((state_root, Report(cycles)))
+    }
+
+    fn prove(
+        &self,
+        config: ConfigKind,
+        state_ssz: Vec<u8>,
+        block_ssz: Vec<u8>,
+        cache_ssz: Vec<u8>,
+        phase_bytes: Vec<u8>,
+    ) -> Result<(Vec<u8>, Self::Proof)> {
+        // Refer to https://0xpolygonhermez.github.io/zisk/getting_started/writing_programs.html#prove
+        // 1. Run cmd for program setup
+        //   cargo-zisk rom-setup -e target/riscv64ima-zisk-zkvm-elf/release/zkvm_guest_zisk
+        // 2. Generate proof
+        //   LIB_EXT=$([[ "$(uname)" == "Darwin" ]] && echo "dylib" || echo "so")
+        //   cargo-zisk prove -e target/riscv64ima-zisk-zkvm-elf/release/zkvm_guest_zisk -i build/input.bin -o ./ -a
+        let zisk_guest_dir = Self::get_guest_dir();
+        let input_path =
+            Self::generate_input_file(config, state_ssz, block_ssz, cache_ssz, phase_bytes)?;
+        let elf_path = zisk_guest_dir.join(ELF_PATH_SUFFIX);
+
+        println!("Running program setup `cargo-zisk rom-setup`...");
+
+        // 1. Run cmd for program setup
+        let setup_output = run_cmd(
+            Command::new("cargo-zisk")
+                .arg("rom-setup")
+                .arg("--elf")
+                .arg(&elf_path)
+                .current_dir(&zisk_guest_dir),
+        )?;
+
+        if !setup_output.0.success() {
+            return Err(anyhow!(
+                "Failed to setup zisk proving. Stderr:\n{}",
+                String::from_utf8_lossy(&setup_output.1)
+            ));
+        }
+
+        // 2. Generate proof
+        println!("Proving `cargo-zisk prove`...");
+
+        let prove_output = run_cmd(
+            Command::new("cargo-zisk")
+                .arg("prove")
+                .arg("--elf")
+                .arg(&elf_path)
+                .arg("--input")
+                .arg(&input_path)
+                .arg("--output-dir")
+                .arg(&zisk_guest_dir)
+                .arg("--aggregation") // indicates that a final aggregated proof should be produced
+                .current_dir(&zisk_guest_dir),
+        )?;
+
+        let (exit_status, output_bytes) = prove_output;
+        let output = String::from_utf8_lossy(&output_bytes);
+
+        if !exit_status.success() {
+            return Err(anyhow!("Failed to generate proof. Stderr:\n{output}"));
+        }
+
+        // Gather back the last set_output() from the VM guest output.
+        // Note: maybe able to read from the output dir the public output.
+        let state_root = Self::collect_result(&output)?;
```

Comment:
I want to keep this line here. Because I have never able to execute to this point actually (**prove** mode). So keeping this line let future dev know they could either read from the output dir (I'm not sure what's the output file name), or collect the result from the screen output.

Reply:
I believe you can just comment out whole guest program, keep empty `main`, and then run prove mode locally, so that you can find where file is located.

It is better to check proving mode before merging, because after preparing all zkvms, we want to conduct some kind of report. So we will likely hand out code to zisk team and ask them to run proving on their rig.